pisces/data_sets.py

```diff
@@ -176,10 +176,14 @@ class IdExtractor(SimplifiablePrefixTree):
     def __init__(self, delimiter: str = "", key: str = ""):
         super().__init__(delimiter, key)
 
-    def extract_ids(self, files: List[str], id_template: str | None=None) -> List[str]:
+    def extract_ids(self, 
+                    files: List[str], 
+                    id_template: str | None=None,
+                    id_symbol: str="<<ID>>") -> List[str]:
```

Comment:
Since this is kind of a general-purpose class, used by DataSetObject, I think we shouldn't have a default here. See my other comment about it being cumbersome long-run.